### PR TITLE
feat: #6 Add codecov action

### PR DESCRIPTION
**Pull request by AlexAxthelm**
Add action that checks codecoverage, and uploads results to codecov.io

Closes: #6

### CODE REVIEW

File: .github/workflows/R-codecov.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+name: codecov
+
+jobs:
+  test-coverage:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v3
```
Comment (cjyetman) [inline .github/workflows/R-codecov.yml:15]:
```suggestion
      - uses: actions/checkout@v4
```

File: .github/workflows/R-codecov.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Show testthat output
+        if: always()
+        run: |
+          ## --------------------------------------------------------------------
+          find ${{ runner.temp }}/package -name 'testthat.Rout*' -exec cat '{}' \; || true
+        shell: bash
+
+      - name: Upload test results
+        if: failure()
+        uses: actions/upload-artifact@v3
```
Comment (cjyetman) [inline .github/workflows/R-codecov.yml:44]:
```suggestion
        uses: actions/upload-artifact@v4
```

File: .github/workflows/R-codecov.yml (hunk 1 above)
Comment (jdhoffa) [reply .github/workflows/R-codecov.yml:15]: See https://github.com/RMI-PACTA/actions/issues/53

File: .github/workflows/R-codecov.yml (hunk 2 above)
Comment (jdhoffa) [reply .github/workflows/R-codecov.yml:44]: FYI. https://github.com/RMI-PACTA/actions/issues/53

**Review by cjyetman — APPROVED** [review]

**Pull request merged**